### DISCUSSION TITLE
相册里效果出不来

### DISCUSSION THREAD

**Discussion by jimmy201516** [MaLuns/vitepress-theme-async#7] [General]
你好，全局组件注册已打开，在使用gallery相册时，法线没办法渲染，可以帮忙看看吗，感谢回复。
![001](https://github.com/user-attachments/assets/4de7c38c-74c3-4854-8148-713f7bd4004d)

**Comment by jimmy201516** [comment]
![002](https://github.com/user-attachments/assets/82144980-522d-47f8-bab8-50b8eb4afe61)

**Reply by MaLuns** [reply]
看看html里组件没有渲染吗

**Reply by jimmy201516** [reply]
谢谢，后面我找到了，与第二个discussion一样的，dev时候出不来，build可以出来

**Reply by MaLuns** [reply]
> 谢谢，后面我找到了，与第二个discussion一样的，dev时候出不来，build可以出来

🤔看了下，通过 npm 安装情况下，似乎没有注册到全局

**Reply by jimmy201516** [reply]
不是很明白哦
不是通过globalComponents: true 注册就可以了吗

**Reply by MaLuns** [reply]
通过 npm 安装方式，目前按照下面方式修改 .vitepress\theme\index.ts  文件

```
import { defineTheme } from 'vitepress-theme-async';
export default defineTheme({
    enhanceApp({ app, siteData }) {
        const globalComponents = siteData.value.themeConfig.globalComponents;
        if (globalComponents) {
            const modulesFiles = import.meta.glob('../../node_modules/vitepress-theme-async/components/global/*.vue', { import: 'default', eager: true });
            console.log(modulesFiles, import.meta)
            const flag = Array.isArray(globalComponents);
            for (const path in modulesFiles) {
                const componentName = path.split('/').reverse()[0].replace(/.vue$/, '');
                if (!flag || (flag && globalComponents.includes(componentName))) {
                    app.component(componentName, modulesFiles[path]);
                }
            }
        }
    },
})

```

**Reply by jimmy201516** [reply]
这个提示没有glob属性

**Comment by jimmy201516** [comment]
摘要好像也出不来呢
`---
title: 摘要测试
date: 2022-09-21 15:08:29
cover: [https://hexo-theme-async.imalun.com/imgs/demo2.png]
sticky: 1
author: 张三
categories: [Wiki]
tags: [材质,工艺]
excerpt: 摘要
---

这里是摘要内容


# 摘要测试`

**Reply by MaLuns** [reply]
摘要用 <!—-more—-> ，会获取 md 里标签前内容

**Reply by jimmy201516** [reply]
是的，md转义了
![2024-11-05_104558](https://github.com/user-attachments/assets/816f9dd8-cc3d-47a1-8997-60eb6b5d0fa8)

**Reply by MaLuns** [reply]
> 是的，md转义了 ![2024-11-05_104558](https://private-user-images.githubusercontent.com/15228709/382988084-816f9dd8-cc3d-47a1-8997-60eb6b5d0fa8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3Nzg0MDEsIm5iZiI6MTczMDc3ODEwMSwicGF0aCI6Ii8xNTIyODcwOS8zODI5ODgwODQtODE2ZjlkZDgtY2MzZC00N2ExLTg5OTctNjBlYjZiNWQwZmE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA1VDAzNDE0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxOTMxNzcyMzlkNGMxMzVlNGJlYTkyOTdiODIxOTM0ZTY1YmI1ZThjNjA0YTQ3YTE3YTc4NTFjMTRhMjU5NzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4IYfSoA3JawRI3ECctGrvcq5Xd5gKnJVacuu7B23wjQ)

测试归档页可以正常显示呀，你是哪个地方摘要不显示

![image](https://github.com/user-attachments/assets/c3bc9cba-07d1-4927-ae47-2d2a8d544eaf)

**Comment by jimmy201516** [comment]
我尝试了重新安装，步骤如下，主题用不上哈
![01](https://github.com/user-attachments/assets/a2478844-7270-45c3-9a5c-c7f8717d830d)
![02](https://github.com/user-attachments/assets/c03e38fb-d6d1-4f35-9025-055ca94b03e0)
![03](https://github.com/user-attachments/assets/d8e8661a-724e-43c5-8438-3cdc4be92369)

**Reply by MaLuns** [reply]
有完整 demo 吗 ，文章目录 `postDir` 设置是啥？首页 index.md 添加了吗？只看这几个图不好判断啥问题

**Comment by jimmy201516** [comment]
请帮忙看看，能看出哪里的问题吗，谢谢

**Comment by jimmy201516** [comment]
1.安装vitepress
npm add -D vitepress

2.初始化vitepress
npx vitepress init
![图片1](https://github.com/user-attachments/assets/ddf4d88a-7f29-4c95-b6f1-07f6b1f037d0)

3.验证正常
![图片2](https://github.com/user-attachments/assets/364eb5ea-71b8-4aab-8d0f-d5483755a193)
![图片3](https://github.com/user-attachments/assets/5d96808c-a67e-4126-bdf6-92317d772018)

4.npm安装theme
![图片4](https://github.com/user-attachments/assets/dba8fac1-7a2f-44e6-ba1c-732d01aebbcc)

安装完成后，查看Node_modules
![图片5](https://github.com/user-attachments/assets/277939c0-5370-41b6-a38b-1e8b1a8289c4)


5.配置主题.vitepress/config.mts
![图片6](https://github.com/user-attachments/assets/f287949f-5081-4841-adb0-927e2b59828c)

6.配置index.ts
![图片7](https://github.com/user-attachments/assets/2ba150e8-5628-4fb3-978e-72883d81b3a7)

7.Run dev
  npm run docs:dev
  
![图片8](https://github.com/user-attachments/assets/d88ab817-ab8a-47f8-99e2-6ab85de066a7)

![图片9](https://github.com/user-attachments/assets/5b7ac403-16fc-49d4-8224-bfea05889572)

查看html源码如下：
![图片10](https://github.com/user-attachments/assets/a7f998a9-8099-4be8-870a-8a4e145e4abd)

**Reply by jimmy201516** [reply]
你好，我这边写了下整个配置的过程，烦请看看，谢谢

**Reply by MaLuns** [reply]
你的 index.md 里 layout 没有设置吧，文章 .md 文件 Frontmatter 内容调整下，可以参考[文档里说明](https://vitepress-theme-async.imalun.com/guide/page#frontmatter) 
![image](https://github.com/user-attachments/assets/6e47e61b-89db-4807-854e-3da863ac1bfc)

**Reply by jimmy201516** [reply]
这样又回到了之前的问题，摘要和相册的问题，摘要出不来，相册只能在build的时候才有

**Comment by jimmy201516** [comment]
npm run docs:build时，报了下面错误，这个需要怎么调整吗，谢谢
![2024-11-06_152433](https://github.com/user-attachments/assets/f25c28be-2c69-4c1b-a474-dd8f2c68a764)

**Reply by jimmy201516** [reply]
能看出这个是哪里的问题吗，看起来是date属性未定义，但我不能定位到问题在哪里

**Reply by MaLuns** [reply]
> npm run docs:build时，报了下面错误，这个需要怎么调整吗，谢谢 ![2024-11-06_152433](https://private-user-images.githubusercontent.com/15228709/383447747-f25c28be-2c69-4c1b-a474-dd8f2c68a764.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4ODQxMjMsIm5iZiI6MTczMDg4MzgyMywicGF0aCI6Ii8xNTIyODcwOS8zODM0NDc3NDctZjI1YzI4YmUtMmM2OS00YzFiLWE0NzQtZGQ4ZjJjNjhhNzY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDA5MDM0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlNTBmNDgwM2E0ODFhMmI2OWFmMWFhOGZiYjQwMjAwZTlkZWFlM2Q1YWQzODNlOTM2MTNiYzBiYTMzZGNmMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0VQU6UxSUfEiCZ48vEYQtvx0Yo4bQ4Ju7DKGFE4OrPE)

不太清楚你操作了啥，我按照你的操作下来 【调整 index.md 里 layout ，文章 .md 文件 Frontmatter 内容调整下】 本地运行和打包都是可以的呀

[demo.zip](https://github.com/user-attachments/files/17644570/demo.zip)

上传了个 demo , 你下载下看看

**Reply by jimmy201516** [reply]
谢谢

**Comment by jimmy201516** [comment]
首先，非常感谢你的帮助，谢谢
使用你提供的demo, 然后copy我这边的config配置及post md我也能build成功了，我也不知道原因^-^,总之就是可以了，我就在你这个demo上去做吧，谢谢
![2024000](https://github.com/user-attachments/assets/b1acab51-e7d2-4d81-87bd-f71d3f890e21)

但是，摘要和画册的问题依旧，可以再麻烦你在demo的基础上验证一下吗，谢谢
我使用demo包还是以下情况
![202401](https://github.com/user-attachments/assets/8fbdd2fb-6546-4122-90fb-98486734bee7)
![202402](https://github.com/user-attachments/assets/314c1a74-6667-4bbc-ba6c-ea3b208fa9ba)
![2024003](https://github.com/user-attachments/assets/7d096809-13ad-49b9-a749-1baadf8f70e2)

**Comment by MaLuns** [comment]
摘要是不是没配置这个

export default defineConfig({
	themeConfig: {
		archiveGenerator: {
			style: 'more', 
		}
	}
});

**Reply by jimmy201516** [reply]
刚刚试过了，摘要就是这里的问题
